wp-includes/class-wp-service-workers.php

```diff
@@ -149,87 +154,95 @@ public function get_current_scope() {
 	}
 
 	/**
-	 * Get offline page.
+	 * Get script for handling of error responses when the user is offline or when there is an internal server error.
 	 *
-	 * @return WP_Post|null Offline page if selected and published, or else null.
+	 * @return string Script.
 	 */
-	protected function get_offline_page() {
-		$offline_page_id = (int) get_option( WP_Offline_Page::OPTION_NAME, 0 );
-		if ( ! $offline_page_id ) {
-			return null;
-		}
-		if ( 'publish' !== get_post_status( $offline_page_id ) ) {
-			return null;
-		}
-		return get_post( $offline_page_id );
-	}
+	public function get_error_response_handling_script() {
 
-	/**
-	 * Add caching for the offline page.
-	 *
-	 * When an offline page has been selected and this page is published, add this offline page to the precache.
-	 * Also include the featured image that the offline page has associated with it, as well as any selected
-	 * background image and header image. Otherwise, enable runtime stale-while-revalidate caching of all other
-	 * theme assets so that they will be available on the offline page.
-	 *
-	 * @todo It may be unsafe to use staleWhileRevalidate caching here. It is needed for caching background images in stylesheets. It may be safer to do networkFirst, at least when SCRIPT_DEBUG is enabled.
-	 */
-	protected function add_offline_page_caching() {
-		$offline_page = $this->get_offline_page();
-		if ( ! $offline_page ) {
-			return;
+		$revision = sprintf( '%s-v%s', get_template(), wp_get_theme( get_template() )->Version );
+		if ( get_template() !== get_stylesheet() ) {
+			$revision .= sprintf( ';%s-v%s', get_stylesheet(), wp_get_theme( get_stylesheet() )->Version );
```

Comment:
Is there a need to call `get_template()` and `get_stylesheet()` multiple times here?  Each time we call it, it's passing through the filters and consuming a bit of processing time.  Could we instead store it into a variable and then reuse the variable here?

Reply:
Yeah, good call. Fixed in 2aed1a9f2a19ac329fa256dff47df8d063df82ae.